zerver/views/streams.py

```diff
@@ -518,27 +532,86 @@ def remove_subscriptions_backend(
                 user_full_name=silent_mention_syntax_for_user(acting_user),
                 stream_name=f"#**{subscriptions[0]}**",
             )
-
-    with override_language(recipient_user.default_language):
-        message = _("{user_full_name} subscribed you to the following streams:").format(
-            user_full_name=silent_mention_syntax_for_user(acting_user),
-        )
+        elif remove_subscription and len(subscriptions) == 1:
+            return _("{user_full_name} unsubscribed you from the stream {stream_name}.").format(
+                user_full_name=silent_mention_syntax_for_user(acting_user),
+                stream_name=f"#**{subscriptions[0]}**",
+            )
+        elif add_subscription:
+            message = _("{user_full_name} subscribed you to the following streams:").format(
+                user_full_name=silent_mention_syntax_for_user(acting_user),
+            )
+        else:
+            message = _("{user_full_name} unsubscribed you from the following streams:").format(
+                user_full_name=silent_mention_syntax_for_user(acting_user),
+            )
     message += "\n\n"
     for stream_name in subscriptions:
         message += f"* #**{stream_name}**\n"
     return message
 
 
+def send_messages_for_removed_subscribers(
+    removed_subscribers: List[Tuple[UserProfile, Stream]], acting_user: UserProfile
+) -> None:
+    if len(removed_subscribers) == 0:
+        return
+
+    user_id_to_unsubscribed_streams: Dict[Any, Set[str]] = {}
+    user_id_to_user: Dict[Any, UserProfile] = {}
+    for user, stream in removed_subscribers:
+        if user.id in user_id_to_unsubscribed_streams:
+            user_id_to_unsubscribed_streams[user.id].add(stream.name)
+        else:
+            user_id_to_unsubscribed_streams[user.id] = {stream.name}
+            user_id_to_user[user.id] = user
+
+    mention_backend = MentionBackend(acting_user.realm_id)
+    all_messages = []
+    for user_id, stream_names in user_id_to_unsubscribed_streams.items():
+        # Our caller already filters deactivated users. We need to check for
+        # other cases where we shouldn't send notification message.
+        if user_id == acting_user.id:
+            continue
+        if user_id_to_user[user_id].is_bot:
+            continue
+
+        sender = get_system_bot(settings.NOTIFICATION_BOT, user_id_to_user[user_id].realm_id)
+        message = make_message_for_subscription_change(
+            acting_user=acting_user,
+            recipient_user=user_id_to_user[user_id],
+            stream_names=stream_names,
+            remove_subscription=True,
+        )
+
+        all_messages.append(
+            internal_prep_private_message(
+                sender=sender,
+                recipient_user=user,
+                content=message,
+                mention_backend=mention_backend,
+            )
+        )
+
+    do_send_messages(all_messages)
```

Comment:
`do_send_messages(message_requests, mark_as_read=[acting_user.id]))`

Reply:
Changed this call as suggested.